ml-algorithms/src/main/java/org/opensearch/ml/engine/algorithms/agent/MLReActAgentRunner.java

```diff
@@ -159,19 +160,20 @@ private void runAgent(MLAgent mlAgent, Map<String, String> params, ActionListene
                     executeParams.put(key.replace(toolSpec.getName()+".", ""), params.get(key));
                 }
             }
-            if (!toolFactories.containsKey(toolSpec.getType())) {
-                listener.onFailure(new IllegalArgumentException("No tool factory found for " + toolSpec.getType()));
-                return;
-            }
-            Tool tool = toolFactories.get(toolSpec.getType()).create(toolParams);
-            tool.setName(toolSpec.getName());
+            try {
+                Tool tool = toolsFactory.getTool(toolSpec.getType());
+                tool.setName(toolSpec.getName());
```

Comment:
ToolSepc name could be null, need if condition check here to avoid override to null.

Reply:
Added null check in `getTool` function.